### DISCUSSION TITLE
implementation of cgi requests in Mongoose ?

### DISCUSSION THREAD

**Discussion by MalcolmGillott** [cesanta/mongoose#2403] [Ask questions, get answers] [answered]
I am experimenting with Mongoose on an embedded linux system that was previously using httpd web server

Mongoose works fine for static content, but we also make use of CGI requests for dynamic content . 

Can Mongoose be configured to handle CGI request directly by assigning a 'cgi-bin' URI ?
 or do I have to make use of the RPC interface or similar ?

Thanks
Mig

**Comment by scaprile** [comment, marked as answer]
See what's covered in our many [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.
https://mongoose.ws/documentation/tutorials/http-server/
https://mongoose.ws/documentation/#web-ui

**Reply by scaprile** [reply]
BTW, we support SSI too

**Reply by smartelements** [reply]
I tried to use cgi scripts in mongoose but after several hours I still get plain text output instead of execution of the scripts. How can I get ti to work in a C program? Thank you

**Reply by scaprile** [reply]
Please no thread hijacking
Please read the docs. If you would have seen at least ONE example you would have saved your several hours. I think you are confusing what Mongoose actually is with that plain executable gamers like for some reason.
